File: backend/StudyFlow.Api/Domain/Entities/Tema.cs

namespace StudyFlow.Api.Domain.Entities
{
    public class Tema
    {
        public int Id { get; set; }
        public string? Nome { get; set; }
        public string? Descricao { get; set; }
        public DateTime DataCriacao { get; set; } = DateTime.UtcNow;

        public Guid UsuarioId { get; set; }
        public Usuario? Usuario { get; set; }

        public ICollection<Nota> Notas { get; set; } = new List<Nota>();
    }
}

File: frontend/src/types/tema.ts
export interface Tema {
  id: number;
  nome: string;
  descricao?: string | null;
  data_criacao?: string | null;
}

export interface TemaRequestDto {
  nome: string;
  descricao?: string | null;
}

export interface TemaResponseDto {
  id: number;
  nome: string;
  descricao?: string | null;
  data_criacao?: string | null;
}
